libdnf/dnf-utils.c

```diff
@@ -118,7 +118,7 @@ dnf_remove_recursive(const gchar *directory, GError **error)
                 return FALSE;
         } else {
             g_debug("deleting file %s", src);
-            if (g_unlink(src) != 0) {
+            if ((g_unlink(src) != 0) && g_file_test(src, G_FILE_TEST_EXISTS)) {
```

Comment:
I think the usual way to do the test is to check the return error code instead. This probably doesn't matter much here, but usually temp directory creation man pages always say to avoid testing for file existence and instead test for return values, in order to avoid races.

Something like this, maybe?

    if (g_unlink (src) != 0 &&
        errno != ENOENT) {
        ...
    }